shared/ui/controlled/controlled-text-field.tsx

```diff
@@ -0,0 +1,42 @@
+import { Control, FieldPath, FieldValues, useController } from 'react-hook-form';
+
+import { assertUnreachable } from '@/shared/helpers/assertUnreachable';
+import { TextField, TextFieldProps } from '@/shared/ui/textField';
+
+type ControllerProps<TFieldValues extends FieldValues> = {
+  control: Control<TFieldValues>;
+  name: FieldPath<TFieldValues>;
+};
+type FilteredTextFieldProps<T extends TextFieldProps> = Omit<
+  T,
+  'error' | 'onBlur' | 'onChange' | 'value'
+>;
+
+// prettier-ignore
+export type ControlledTextFieldProps<TFieldValues extends FieldValues> =
+  | (ControllerProps<TFieldValues> &
+      FilteredTextFieldProps<Extract<TextFieldProps, { as: 'textarea' }>>)
+  | (ControllerProps<TFieldValues> &
+      FilteredTextFieldProps<Extract<TextFieldProps, { as?: 'input' }>>);
+
+export const ControlledTextField = <TFieldValues extends FieldValues = FieldValues>({
+  control,
+  disabled,
+  name,
+  ...props
+}: ControlledTextFieldProps<TFieldValues>) => {
+  const {
+    field,
+    fieldState: { error },
+  } = useController({ control, disabled, name });
```

Comment:
We need to pass `disabled` to `useController` to exclude the current field from form validation. There is no need to pass `disabled` to `Textfield` manually, because the `field` object returned by the `useController` call will already have this value.